File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/postal/SaveWktblPostalCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.postal;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.DtoEntityInitialValueInterface;
import net.seijishikin.jp.normalize.common_tool.dto.FrameworkCapsuleDto;
import net.seijishikin.jp.normalize.manage.kanrensha.entity.WkTblPostalEditEntity;

/**
 * 郵便番号更新内容Dto
 */
public class SaveWktblPostalCapsuleDto extends FrameworkCapsuleDto // NOPMD DataClass
        implements Serializable, DtoEntityInitialValueInterface {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 郵便番号差分ワークテーブルEntity */
    private WkTblPostalEditEntity editEntity = new WkTblPostalEditEntity();

    /** 削除該否 */
    private Boolean isDelete = INIT_BOOLEAN;

    /**
     * 郵便番号差分ワークテーブルEntityを取得する
     * 
     * @return 郵便番号差分ワークテーブルEntity
     */
    public WkTblPostalEditEntity getEditEntity() {
        return editEntity;
    }

    /**
     * 郵便番号差分ワークテーブルEntityを設定する
     * 
     * @param editEntity 郵便番号差分ワークテーブルEntity
     */
    public void setEditEntity(final WkTblPostalEditEntity editEntity) {
        this.editEntity = editEntity;
    }

    /**
     * 削除該否を取得する
     * 
     * @return 削除該否
     */
    public Boolean getIsDelete() {
        return isDelete;
    }

    /**
     * 削除該否を設定する
     * 
     * @param isDelete 削除該否
     */
    public void setIsDelete(final Boolean isDelete) {
        this.isDelete = isDelete;
    }

}

File: app/front/src/components/main/dto/address_postal/saveWktblPostalCapsuleDto.ts
import { FrameworkCapsuleDto, type FrameworkCapsuleDtoInterface } from "seijishikin-jp-normalize_common-tool";
import { WkTblPostalEditEntity, type WkTblPostalEditEntityInterface } from "../../entity/wkTblPostalEditEntity";

interface SaveWktblPostalCapsuleDtoInterface extends FrameworkCapsuleDtoInterface {

    /** 郵便番号差分ワークテーブルEntity */
    editEntity: WkTblPostalEditEntityInterface;

    /** 削除該否 */
    isDelete: boolean;
}

class SaveWktblPostalCapsuleDto extends FrameworkCapsuleDto implements SaveWktblPostalCapsuleDtoInterface {


    /** 郵便番号差分ワークテーブルEntity */
    editEntity: WkTblPostalEditEntityInterface;

    /** 削除該否 */
    isDelete: boolean;

    constructor() {
        super();

        this.editEntity = new WkTblPostalEditEntity();
        this.isDelete = false;
    }
}

export { type SaveWktblPostalCapsuleDtoInterface, SaveWktblPostalCapsuleDto }
